### PR TITLE
Improve README.md

### PR DESCRIPTION
**Pull request by mauricioirace**
Updating documentation.
Adding more detailed information on running relay server

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,19 +22,208 @@ Our main objective is to provide the RSK ecosystem with the means to enable bloc
...
+logs you will be seing a lot of interaction but you need to see this log to be sure the server is ready:
+
+```
+Relayer state: READY
+```
+
+### On Testnet
+
+1. Send to relayManagerAddress at least 0.001 tRBTC to set it up
+2. Send to relayWorkerAddress at least 0.001 tRBTC to set it up
```
Comment (mortelli) [inline README.md:212]: we should indicate how to obtain these addresses, (i.e. go to `http://localhost:8080/`or whatever the URL is)

File: README.md [hunk 2]
```diff
@@ -22,19 +22,208 @@ Our main objective is to provide the RSK ecosystem with the means to enable bloc
...
 It is important to recall that - as a security measure - the version 1 contracts deployed on Mainnet have limits on the staked amounts to operate, these limits were removed in version 2.
 
-## Technical Documentation
+## Table of contents
+
+- [Requirements](#Requirements)
+- [Launching](#Launching)
+- [Testnet Contracts](#Testnet-Contracts-V2)
+- [Troubleshooting](#Troubleshooting)
+- [Changelog](#Changelog)
```
[the remark this answers is not in the document]
Comment (JONAF2103) [reply README.md]: agree

File: README.md [hunk 3]
```diff
@@ -22,19 +22,208 @@ Our main objective is to provide the RSK ecosystem with the means to enable bloc
...
+## Additional Technical Documentation
 
 The following technical content is available:
 
 - Architecture [docs/enveloping_architecture](docs/enveloping_architecture.md)
-- Installing requirements [docs/basic_requirements](docs/basic_requirements.md)
 - Launching RIF Enveloping [docs/launching_enveloping](docs/launching_enveloping.md)
 - Development guide [docs/development_guide](docs/development_guide.md)
 - Integration guide [docs/integration_guide](docs/integration_guide.md)
 - Gas costs [docs/overhead_tx_costs](docs/overhead_tx_costs.md)
```
[the remark this answers is not in the document]
Comment (JONAF2103) [reply README.md]: Agree here too, it's better to have the link directly

File: README.md [hunk 4]
```diff
@@ -22,19 +22,208 @@ Our main objective is to provide the RSK ecosystem with the means to enable bloc
...
+```
+
+You need to copy that summary and save it in some place to retrieve it later.
+
+### On Testnet
+
+
+Use `truffle` for deploying contracts.
+
+`npx truffle migrate --network rsktestnet` (disclaimer: to use testnet, you should have an unlocked account with funds or configure it in `truffle.js`).
```
[the remark this answers is not in the document]
Comment (JONAF2103) [reply README.md:141]: sounds good

File: README.md [hunk 5]
```diff
@@ -22,19 +22,204 @@ Our main objective is to provide the RSK ecosystem with the means to enable bloc
...
+  "port": 8090,
+  "relayHubAddress": "0x3bA95e1cccd397b5124BcdCC5bf0952114E6A701",
+  "relayVerifierAddress": "0x74Dc4471FA8C8fBE09c7a0C400a0852b0A9d04b2",
+  "deployVerifierAddress": "0x1938517B0762103d52590Ca21d459968c25c9E67",
+  "gasPriceFactor": 1,
+  "rskNodeUrl": "http://rsk-node:4444",
+  "devMode": true,
+  "customReplenish": false,
+  "logLevel": 1,
+  "workdir": "/home/jonathan/workspace/enveloping/environment"
```
Comment (JONAF2103) [inline README.md:160]: remove my name please lol

File: README.md [hunk 6]
```diff
@@ -22,19 +22,204 @@ Our main objective is to provide the RSK ecosystem with the means to enable bloc
...
+first need to configure the script located on `<PROJECT_ROOT>/scripts/registerRelayServer` and replace the 
+   values as you consider. The script contains the following:
+
+```
+node dist/src/cli/commands/enveloping.js relayer-register --funds 100 --stake 200 --network http://rsk-node:4444/ --hub "0x3bA95e1cccd397b5124BcdCC5bf0952114E6A701"
+```
+
+Where:
+
+* **--funds**: indicates the amount of RBTC that you will transfer from accounts[0] to the worker manager account.
```
Comment (JONAF2103) [inline README.md:195]: You should specify the `--network` parameter here too

File: README.md (hunk 1 above)
Comment (JONAF2103) [reply README.md:210]: agree

File: README.md [hunk 7]
```diff
@@ -22,19 +22,204 @@ Our main objective is to provide the RSK ecosystem with the means to enable bloc
...
+
+## Custom worker replenish function in the Relay Server
+
+Each relayed transaction is signed by a Relay Worker account. The worker accounts are controlled by the Relay Manager. When a relay worker signs and relays a transaction, the cost for that transaction is paid using the funds in that worker's account. If the transaction is not subsidized, then the worker is compensated with tokens. Worker accounts must always have some minimum balance to pay gas for the transaction. These balances can be managed by implementing a replenishment strategy. The Relay Manager can use the strategy to top off a relay worker's account when the balance gets too low. We provide a default implementation for a replenishment strategy.  Enveloping solution integrators can implement their own replenish strategy.
+
+To implement and use your own replenish strategy:
+
+1. In the folder `src/relayserver`, open `ReplenishFunction.ts` with a text editor.
+2. On the function `replenishStrategy` write your replenish strategy on the then branch.
```
Comment (JONAF2103) [inline README.md:220]: This should be 
```suggestion
2. On the function `replenishStrategy` write your new replenish strategy.
```

File: README.md [hunk 8]
```diff
@@ -3,6 +3,26 @@
...
 [![CircleCI](https://circleci.com/gh/rsksmart/enveloping/tree/master.svg?style=shield)](https://circleci.com/gh/rsksmart/enveloping/tree/master)
+
+
+## Table of contents
+
+- [Description](#Description)
+- [Requirements](#Requirements)
+- [Launching](#Launching)
+- [Testnet Contracts](#Testnet-Contracts-V2)
+- [Troubleshooting](#Troubleshooting)
```
Comment (mortelli) [inline README.md:14]: broken link here

File: README.md [hunk 9]
```diff
@@ -22,19 +22,208 @@ Our main objective is to provide the RSK ecosystem with the means to enable bloc
...
+After this step, you must make sure that your `PATH` variable gives priority to `/usr/local/bin` over `/usr/bin`. You can check this with `which readlink`, which should output `/usr/local/bin/readlink`. Alternatively try executing `readlink -f .`, if it works you're ok.
+
+# Launching
+
+## Building the project
+
+Clone the project. Then run the following from the Enveloping project's root directory to build it.
+
+`yarn install`
+`yarn prepare`
```
[the remark this answers is not in the document]
Comment (mortelli) [reply README.md:101]: this one is still pending

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (mortelli) [reply README.md:139]: @mauricioirace

File: README.md [hunk 10]
```diff
@@ -22,19 +42,233 @@ Our main objective is to provide the RSK ecosystem with the means to enable bloc
...
+  "port": 8090,
+  "relayHubAddress": "0x3bA95e1cccd397b5124BcdCC5bf0952114E6A701",
+  "relayVerifierAddress": "0x74Dc4471FA8C8fBE09c7a0C400a0852b0A9d04b2",
+  "deployVerifierAddress": "0x1938517B0762103d52590Ca21d459968c25c9E67",
+  "gasPriceFactor": 1,
+  "rskNodeUrl": "http://rsk-node:4444",
+  "devMode": true,
+  "customReplenish": false,
+  "logLevel": 1,
+  "workdir": "/home/user/workspace"
```
Comment (mortelli) [inline README.md:160]:
```suggestion
  "workdir": "/home/user/workspace/enveloping"
```

**Review by mortelli — APPROVED** [review]
minor corrections only :+1:

**Review by JONAF2103 — APPROVED** [review]
LGTM you can move it to RFR

**Pull request merged**